Modix.Services.Test/MediatrIntegrationTests.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Threading.Tasks;
+using Discord;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+using Modix.Services.Core;
+using Modix.Services.Messages.Discord;
+using Modix.Services.Moderation;
+using Moq;
+using Moq.AutoMock;
+using NUnit.Framework;
+
+namespace Modix.Services.Test
+{
+    public class MediatrIntegrationTests
+    {
+        [Test]
+        public async Task MediatrCanResolveNotificationHandlers()
+        {
+            var autoMocker = new AutoMocker();
+            var sut = autoMocker.CreateInstance<ModerationInvitePurgingHandler>();
+
+            var services = new ServiceCollection()
+                .AddMediator()
+                .AddSingleton<INotificationHandler<ChatMessageReceived>, ModerationInvitePurgingHandler>(p => sut)
+                .BuildServiceProvider();
+
+            var mediator = services.GetService<IMediator>();
+            var msg = MockMessage("https://discord.gg/invite/asdf");
+            await mediator.Publish(msg);
+
+            autoMocker.GetMock<IModerationService>().Verify(m => m.DeleteMessageAsync(msg.Message, It.IsAny<string>()));
```

Comment:
This might be obnoxiously picky, but...

You should make an alias extension method for Verify, named `ShouldHaveReceived()`. This keeps the pattern that `Shouldly` uses where any actual Assertion has the word `Should` in it, so they can be easily spotted. I did this for the `NSubstitute` assertion methods in `Modix.Data.Test`.